clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1290,7 +1290,7 @@ public void handleResponse(AbstractResponse response) {
             } else if (error == Errors.NOT_COORDINATOR || error == Errors.COORDINATOR_NOT_AVAILABLE) {
                 lookupCoordinator(FindCoordinatorRequest.CoordinatorType.TRANSACTION, transactionalId);
                 reenqueue();
-            } else if (error == Errors.COORDINATOR_LOAD_IN_PROGRESS || error == Errors.CONCURRENT_TRANSACTIONS) {
+            } else if (error == Errors.COORDINATOR_LOAD_IN_PROGRESS || error == Errors.CONCURRENT_TRANSACTIONS || error == Errors.REQUEST_TIMED_OUT) {
```

Comment:
I'm wondering if we should be even more lenient with the error codes and check for `RetriableException`?

Reply:
Also, can we check the other handlers? We may as well update all of them.

Reply:
I think this is only for the init producer ID since that is the only request that also goes to the controller (the initProducerIdRequest now needs to make a second request to the controller)

I can double check though.

Reply:
Yeah, I'm suggesting that we check the timeout error code (or retriable errors generally) for all of the handlers. That protects the client from future changes on the broker if the error code is not used today.

Reply:
I've updated the handlers. Please take a look. I tried to make sure that the errors I replaced in the line were all retriable and all the ones below were not, but it would be good to have a second pair of eyes check.